sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2657,6 +2657,29 @@ private <T> Flux<FeedResponse<Document>> createPointReadOperations(
                         BridgeInternal.getClientSideRequestStatics(cosmosItemResponse.getDiagnostics())));
 
                 return Mono.just(feedResponse);
+            })
+            .onErrorResume(throwable -> {
+                if (throwable instanceof CosmosException) {
+                    CosmosException cosmosException = (CosmosException) throwable;
+
+                    int statusCode = cosmosException.getStatusCode();
+                    CosmosDiagnostics diagnostics = cosmosException.getDiagnostics();
+
+                    switch (statusCode) {
+                        case HttpConstants.StatusCodes.NOTFOUND:
```

Comment:
Is it valid to swallow 404 with any substatuscode?

Reply:
Good catch! The point-read should swallow for 404 / 0 scenarios. For 404 / 1002 scenarios it should retry and bubble up when retries are exhausted so shouldn't be swallowed anywhere in the reactor pipeline.